Republik.ch.js

```diff
@@ -0,0 +1,388 @@
+{
+	"translatorID": "b4c83513-921f-474b-adc3-7017ea690ba7",
+	"label": "Republik.ch",
+	"creator": "Johannes Wüthrich",
+	"target": "^https?://www\\.republik\\.ch/[0-9]*/[0-9]*/[0-9]*/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
+	"lastUpdated": "2021-08-04 20:31:29"
+}
+
+/*
+Republik.ch Translator
+Copyright (C) 2021 Johannes Wüthrich
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+function detectWeb(doc, url) {
+	// TODO match different types of articles / debates / updates / blog-posts / newsletters etc.
```

Comment:
Just matching articles is fine. If possible, you should try to exclude blog posts, newsletters, that kind of thing, since few people will want to add them to Zotero and Embedded Metadata should do a fine enough job when they do.

Reply:
All content featuring pages are actually structured in a very similar way (speaking from a HTML point of view).
Therefore this translator should work on any content featuring page, be it an article, a newsletter or a discussion page.
I already filter out all non-content pages, via a restrictive target directive: ^https?://www\.republik\.ch/[0-9]*/[0-9]*/[0-9]*/

For comparison:
- This is an article: https://www.republik.ch/2021/08/05/dieser-stoff-ist-ein-einziges-minenfeld
- This a newsletter: https://www.republik.ch/2021/08/05/7-uhr-newsletter
- This a debate: https://www.republik.ch/2021/08/04/auf-welche-privilegien-wuerden-sie-zugunsten-der-natur-verzichten/diskussion

The translator extracts the available information in all of these cases.

Filtering out non articles would actually be quite tedious, as it would mean filtering based on some content. In the structure of the page there is not flag / indication if it is an article or not.